packages/common/container-definitions/src/audience.ts

```diff
@@ -21,6 +20,56 @@ export interface IAudienceOwner extends IAudience {
 	 * @returns if a client was removed from the audience
 	 */
 	removeMember(clientId: string): boolean;
+
+	/**
+	 * Notifies Audience that current clientId has changed.
+	 * See {@link IAudience.self} and {@link IAudienceEvents}'s "selfChanged" event for more details.
+	 */
+	setCurrentClientId(clientId: string | undefined): void;
+}
+
+/**
+ * @public
+ */
+export interface IAudienceEvents extends IEvent {
+	(
+		event: "addMember" | "removeMember",
+		listener: (clientId: string, client: IClient) => void,
+	): void;
+	/**
+	 * Notifies that client established new connection and caught-up on ops.
+	 * Values returned by {@link IAudience.self} represent newly established connection.
+	 * and {@link IAudienceEvents}'s "selfChanged" event for more details.
+	 */
+	(
+		event: "selfChanged",
+		listener: (oldClientId: string | undefined, clientId: string) => void,
+	): void;
+}
+
+/**
+ * Return type of {@link IAudience.self}. Please see remarks for {@link IAudience.self} to learn more details on promises.
+ * @public
+ */
+export interface ISelf {
+	/**
+	 * clientId of current or previous connection (if client is in disconnected or reconnecting / catching up state)
+	 * undefined if this client has never connected to the ordering service.
+	 * It changes only when client has reconnected, caught up with latest ops and certain other criteria are met.
+	 */
+	clientId: string | undefined;
```

Comment:
Did you consider making these fields non-optional, but instead having `getSelf(): ISelf | undefined`?